### PR TITLE
Add autoschema package to automatically generate schemas using reflection

### PR DESCRIPTION
**Pull request by akrennmair**
This PR adds an `autoschema` package that uses reflection to automatically generate schemas. The idea is to be eventually fully compatible with `floor`'s reflection marshaller/unmarshaller, but even though I added some tests to validate this, validation is incomplete so far.

### CODE REVIEW

File: parquetschema/autoschema/gen.go [hunk 1]
```diff
@@ -0,0 +1,399 @@
...
+			SchemaElement: &parquet.SchemaElement{
+				Type:           parquet.TypePtr(parquet.Type_BOOLEAN),
+				Name:           fieldName,
+				RepetitionType: parquet.FieldRepetitionTypePtr(parquet.FieldRepetitionType_REQUIRED),
+			},
+		}, nil
+	case reflect.Int:
+		return &parquetschema.ColumnDefinition{
+			SchemaElement: &parquet.SchemaElement{
+				Type:           parquet.TypePtr(parquet.Type_INT32),
```
Comment (panamafrancis) [inline parquetschema/autoschema/gen.go:73]: are we sure int should map to int32 and not int64?

File: parquetschema/autoschema/gen.go [hunk 2]
```diff
@@ -0,0 +1,399 @@
...
+			return &parquetschema.ColumnDefinition{
+				SchemaElement: &parquet.SchemaElement{
+					Type:           parquet.TypePtr(parquet.Type_FIXED_LEN_BYTE_ARRAY),
+					Name:           fieldName,
+					RepetitionType: parquet.FieldRepetitionTypePtr(parquet.FieldRepetitionType_REQUIRED),
+					TypeLength:     &typeLen,
+				},
+			}, nil
+		}
+		// TODO: implement this
```
Comment (panamafrancis) [inline parquetschema/autoschema/gen.go:255]: WDYT of adding the missing types here? at least int/int16/int32/int64

File: floor/writer_test.go [hunk 3]
```diff
@@ -568,12 +567,12 @@ func (r *marshTestRecord) MarshalParquet(obj interfaces.MarshalObject) error {
...
 	}
 
-	fmt.Printf("marshal data: %s", spew.Sdump(obj.GetData()))
+	//fmt.Printf("marshal data: %s", spew.Sdump(obj.GetData()))
 	return nil
 }
 
 func (r *marshTestRecord) UnmarshalParquet(obj interfaces.UnmarshalObject) error {
-	fmt.Printf("unmarshal data: %s", spew.Sdump(obj.GetData()))
+	//fmt.Printf("unmarshal data: %s", spew.Sdump(obj.GetData()))
```
Comment (panamafrancis) [inline floor/writer_test.go:575]: do we need to comment these log statements out?

**Review by panamafrancis — CHANGES_REQUESTED** [review]

File: parquetschema/autoschema/gen.go (hunk 1 above)
Comment (akrennmair) [reply parquetschema/autoschema/gen.go]: Changed to int64.

File: parquetschema/autoschema/gen.go (hunk 2 above)
Comment (akrennmair) [reply parquetschema/autoschema/gen.go]: I looked at what we're doing in `floor`, and essentially, we're treating Go arrays the same as Go slices, so I just unified the code here and treat arrays as if they were slices, even though Go arrays don't translate particularly well to any parquet types.

File: floor/writer_test.go (hunk 3 above)
Comment (akrennmair) [reply floor/writer_test.go]: Removed.

**Review by panamafrancis — APPROVED** [review]

**Pull request merged**